src/extension.ts

```diff
@@ -79,23 +79,68 @@ let decorationType: TextEditorDecorationType = window.createTextEditorDecoration
   }
 );
 
+const jdk8 = "1.8.232";
+const jdk8Action = "JDK8";
+
+const jdk11 = "1.11";
+const jdk11Action = "JDK11";
+
+const default_jabba_ver = "0.11.2";
+
+const config = workspace.getConfiguration("metals");
+
+const java9 = "Java9";
```

Comment:
Suggestion: I would name this constant something more telling, such as `unsupportedJavaVersion9` or something like that.

Also, I would use the same constant where you raise the error, so that if you change it in the future you only have one source of truth for the underlying string.

Reply:
as an aside, did I mention that I really really dislike the `Promise` API because it doesn't carry the error type? Nothing related to this PR, feel free to ignore my rants :D

Reply:
Renamed. I understand why you dislike it, but on the other hand it makes chaining futures much clearer to write and read :D

Reply:
Oh, I'm all for the chaining API, I'm just bothered by the specific implementation :P 

I may propose a rewrite using `TaskEither` from fp-ts in the future ;)